ui/js/controllers/perf/graphs.js

```diff
@@ -421,6 +422,38 @@ perf.controller('GraphsCtrl', [
                     updateDocument();
                     highlightDataPoints();
                 });
+
+                // This part of code is simply copy from overview-plot
+                $('#graph').bind("plotunselected", function() {
+                    $scope.zoom = {};
+                    $scope.selectedDataPoint = null;
+                    hideTooltip();
+                    updateDocument();
+                    plotGraph();
+                });
+                $('#graph').bind("plotselected", function(event, ranges) {
+                    deselectDataPoint();
+                    hideTooltip();
+
+                    $.each($scope.plot.getXAxes(), function(_, axis) {
+                        var opts = axis.options;
+                        opts.min = ranges.xaxis.from;
+                        opts.max = ranges.xaxis.to;
+                    });
+                    $.each($scope.plot.getYAxes(), function(_, axis) {
+                        var opts = axis.options;
+                        opts.min = ranges.yaxis.from;
+                        opts.max = ranges.yaxis.to;
+                    });
+
+                    $scope.zoom = {'x': [ranges.xaxis.from, ranges.xaxis.to], 'y': [ranges.yaxis.from, ranges.yaxis.to]};
+                    $scope.plot.setupGrid();
+                    $scope.plot.draw();
+                    updateDocument();
```

Comment:
Likewise, this is *almost* the same as the plotselected function above, except for the last few lines.

You can do a similar strategy as above by defining a plotSelected function. To get the extra behaviour for the main graph event, do something like this:

```
$('#graph').bind("plotselected", function(event, ranges) {
  plotSelected(event, ranges);
  // Make overview graph display the selected range.
  $scope.plot.clearSelection();
  zoomGraph();
});